src/Microsoft.AspNetCore.Routing/Tree/UrlMatchingNode.cs

```diff
@@ -18,6 +18,8 @@ public UrlMatchingNode(int length)
 
         public int Length { get; }
 
+        public bool IsCatchAll { get; set; }
```

Comment:
This node has information of its 'next' nodes, like if they are literal or catch all etc, but does not have information about itself. We need this information to find if the current node is a catch all so that we can avoid the request path segment length check.